src/sentry/plugins/interfaces/releasehook.py

```diff
@@ -88,5 +89,26 @@ def finish_release(self, version, **values):
             datetime=values['date_released'],
         )
 
+        # check if user exists, and then try to get refs based on version
+        if values.get('owner', None):
```

Comment:
hmm now that i'm looking at it, this should probably all go in the `HerokuReleaseHook`. I think probably the easiest thing do do would be to define a `set_refs` method on the release hook, and then override it in the `HerokuReleaseHook`. So, to do that, you could just add this to the `ReleaseHook` class:

```python
def set_refs(self, release, **values):
    pass
```
and then something like this to `HerokuReleaseHook`:

```python
def set_refs(self, release, **values):
    # check if user exists, and then try to get refs based on version
    if values.get('owner', None):
        try:
            repository = Repository.objects.get(
                organization_id=self.project.organization_id,
                name=ProjectOption.objects.get_value(
                    project=self.project,
                    key='heroku:repository'
                )
            )
        except Repository.DoesNotExist:
            pass
        else:
            release.set_refs(
                refs=[{
                    'commit': release.version,
                    'repository': repository.name}],
                user=values['owner'],
                fetch=True
            )
```
and then call `self.set_refs(release, **values)` here. this will also be good, because then if we ever wanted to do this for freight or another plugin, we'll just be able to update sentry-plugins.